File: frontend/src/api/signals.ts
import { fetchApi } from './client';

export interface SignalItem {
  id: string;
  entity_id: string;
  company_name: string;
  category: string;
  severity: string;
  title: string;
  summary: string;
  quote?: string;
  source_url?: string;
  detected_at: string;
  score: number;
  evidence: Record<string, any>;
}

export interface EUDRChecklistItem {
  item: string;
  status: string;
  article: string;
  gap_detail?: string;
}

export interface SignalListResponse {
  signals: SignalItem[];
  total_count: number;
  eudr_scorecard: {
    entity: string;
    readiness_score: number;
    status: string;
    requirements: EUDRChecklistItem[];
    top_gap: string;
    recommended_action: string;
  };
  freight_benchmark: {
    origin_port: string;
    destination_port: string;
    mode: string;
    container_type: string;
    rate_usd: number;
    rate_spread: string;
    transit_days: string;
    port_congestion_index: string;
    reroute_risk_notes?: string;
    sample_air_transit: string;
  };
}

export const getSignals = () => fetchApi<SignalListResponse>('/api/v1/signals');

File: backend/app/api/signals.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session
from typing import Optional
from app.database import get_db
from app.api.deps import require_api_key
from app.repositories import signal_repo, capability_repo
from app.services import compliance_service, lane_service
from app.schemas.signal import SignalListResponse, SignalItem

router = APIRouter(prefix="/api/v1", tags=["Signals"])

@router.get("/signals", response_model=SignalListResponse)
def get_signals_feed(
    category: Optional[str] = Query(None),
    limit: int = Query(50, ge=1, le=100),
    db: Session = Depends(get_db),
    api_key: str = Depends(require_api_key)
):
    signals = signal_repo.get_signals(db, category=category, limit=limit)
    exporter = capability_repo.get_exporter_capability(db)
    eudr = compliance_service.calculate_eudr_readiness(exporter)
    lane = lane_service.get_active_lane_benchmark(db)

    signal_items = []
    for s in signals:
        signal_items.append(SignalItem(
            id=str(s.id),
            entity_id=str(s.entity_id),
            company_name=s.company.canonical_name if s.company else "Global",
            category=s.category,
            severity=s.severity,
            title=s.title,
            summary=s.summary,
            quote=s.quote,
            source_url=s.source_url,
            detected_at=s.detected_at.isoformat() if s.detected_at else "",
            score=float(s.score),
            evidence=s.evidence or {}
        ))

    return SignalListResponse(
        signals=signal_items,
        total_count=len(signal_items),
        eudr_scorecard=eudr,
        freight_benchmark=lane
    )
